### DISCUSSION TITLE
Unable to prepare setup for platform huawei_solar.sensor

### DISCUSSION THREAD

**Discussion by ontijmt** [wlcrs/huawei_solar#271] [Setting up the integration] [answered]
Huawei_solar installed in RBPi 3, but I get this error in HA:

Logger: homeassistant.setup
Source: setup.py:320
First occurred: 12:20:48 (1 occurrences)
Last logged: 12:20:48

Unable to prepare setup for platform huawei_solar.sensor: Platform not found (cannot import name 'UnitOfElectricCurrent' from 'homeassistant.const' (/usr/src/homeassistant/homeassistant/const.py)).

HA System Log file:
Modbus client is not connected to the inverter.
18:44:49 – (ERROR) /usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py - the message first occurred at 12:27:48 and appears 79 times
Error fetching HV2180145294_data_update_coordinator data: Could not update HV2180145294 values: Modbus client is not connected to the inverter.
18:44:19 – (ERROR) Huawei Solar (custom integration) - the message first occurred at 12:27:48 and appears 56 times
Could not read register value, has another device interrupted the connection?
18:33:27 – (ERROR) Huawei Solar (custom integration) - message first occurred at 13:06:23 and appears 25 times

**Comment by wlcrs** [comment, marked as answer]
What version of HA are you running? Only the latest version is supported.

**Reply by ontijmt** [reply]
Home Assistant 2022.10.4
Frontend 20221010.0 - latest
Thanks!

**Reply by wlcrs** [reply]
You'll need to update

**Reply by ontijmt** [reply]
Great! It works now!
Thank you!
